src/main/java/uk/nhs/digital/nhsconnect/nhais/model/fhir/ParametersExtension.java

```diff
@@ -74,7 +76,8 @@ public String extractValue(String name) {
     public Optional<String> extractOptionalValue(String name) {
         return Optional.ofNullable(parameters.getParameter(name))
             .map(StringType.class::cast)
-            .map(StringType::getValueAsString);
+            .map(StringType::getValueAsString)
+            .filter(s -> !StringUtils.isBlank(s));
```

Comment:
Not sure about this. Would we not be silently ignoring a blank freetext e.g. "    "? Would it be better for the segment class to validate isBlank and throw an exception instead?

Reply:
It is not silent ignoring, if the returned optional will be empty, then the FhirValidationException is thrown.
 public String extractValue(String name) {
        return extractOptionalValue(name)
            .orElseThrow(() -> new FhirValidationException("Value " + name + " is missing in FHIR Parameters"));
    }

Reply:
It could be better to do it in Segment, I will check it

Reply:
This class/method shouldn't do any validation on FHIR data (besides null checks to avoid NPE)

Reply:
It doesn't do this anymore. The segment validation is called